playbooks/roles/caasp4/tasks/_assert_packages.yml

```diff
@@ -0,0 +1,75 @@
+---
+#
+# (c) Copyright 2019 SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+- name: Identify localhost SUSE OS distribution and version
+  set_fact:
+    suse_repo_version: "CAASP40-{{ ansible_distribution }}-{{ ansible_distribution_version | replace('.', '') }}"
```

Comment:
instead of defining a var, and checking everything that the var is defined, why wouldn't you include a series of tasks per OS?

An include_tasks with_first_found depending on {{ ansible_distribution }}-{{ ansible_distribution_version }} with a skip: true should do the trick. This will be skipped if nothing is found specific for an os, and will not cause a whole lot of tasks to be evaluated for skipping (only one).

Reply:
Are you asking to have set of tasks for each OS version ?

There are only 2 flows here. 1) If we have OS version specific caasp repo identified, then we install the package
2) If repo is not defined, then we expect user to install manually. We just test for presence of binaries in that case.

But in both cases, we need to verify that installed terraform version is matching v0,11.x

I looked into https://docs.ansible.com/ansible/latest/plugins/lookup/first_found.html and https://docs.ansible.com/ansible/latest/user_guide/playbooks_conditionals.html#selecting-files-and-templates-based-on-variables and did not find any simpler way of following for above logic. 

Logic become quite convoluted as I need to include_task based on first_found in list of task files (install_packages, assert_packages). And then check for terraform version match in both cases.

I will move the code to new role for now.